File: UI/src/app/entities/branch-entity.ts
import {Duration, Moment} from 'moment';

export interface BranchEntity {
  id: string;
  name: string;
  createdAt: Moment | null;
  deletedAt: Moment | null;
  releaseLifetime: Duration | null;
  latestReleaseLifetime: Duration | null;
  useDefaultReleaseLifetime: boolean;
}

File: AppReleases.DataAccess/Entities/BranchEntity.cs
using System.ComponentModel.DataAnnotations;

namespace AppReleases.DataAccess.Entities;

public class BranchEntity
{
    public Guid BranchId { get; init; }
    public Guid ApplicationId { get; init; }
    [MaxLength(32)] public required string Name { get; init; }
    public required DateTime CreatedAt { get; init; }
    public DateTime? DeletedAt { get; init; }
    public TimeSpan? Duration { get; init; }

    public virtual ICollection<ReleaseEntity> Releases { get; init; } = [];
    public virtual ApplicationEntity Application { get; init; } = null!;
}